### DISCUSSION TITLE
How can I do whole mesh smooth with fair filter?

### DISCUSSION THREAD

**Discussion by fgbxx** [pmp-library/pmp-library#193] [Q&A]
Hi, dear all,
I want to smooth a mesh and keep the volume as possible , but If I input a closed whole mesh to the PMP's fair funciton, it return "Missing boundary constraints". So how can I do whole mesh smooth use the fair funciton? Anyone can help?

**Comment by dsieger** [comment]
The `implicit_smoothing()` function with re-scaling enabled (the default) is probably your best shot, see the [API docs](https://www.pmp-library.org/group__algorithms.html#gad0e259304277c0853b3c1867b2bd8eab)

**Reply by fgbxx** [reply]
@dsieger thank you for advice.
But I tried the method on the vessel surface smooth. For the re scale enabled, the vessel shift from the original one.  If I close the re-scaling, also , the tiny vessel will be shrinked.


![res](https://github.com/pmp-library/pmp-library/assets/34882902/0906b4a9-a8c1-4182-8322-05dd3066fadd)
